### PR TITLE
[13.0][ADD] account_invoice_warn_message

### PR DESCRIPTION
**Pull request by HviorForgeFlow**
Account Invoice Warn Message
========================

This module add a warning popup on invoice to ensure warning is populated
no only when partner is changed.

CC @ForgeFlow

### CODE REVIEW

**Review by AdriaGForgeFlow — APPROVED** [review]
Code review :+1:

**Review by bodedra — APPROVED** [review]
LGTM :+1:

File: account_invoice_warn_message/views/account_move.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2020 ForgeFlow S.L.
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record model="ir.ui.view" id="view_move_form">
+        <field name="name">account.move.form (in account_invoice_populate_warn)</field>
```
Comment (MiquelRForgeFlow) [inline account_invoice_warn_message/views/account_move.xml]:
```suggestion
        <field name="name">account.move.form (account_invoice_warn_message)</field>
```

File: account_invoice_warn_message/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+
+    _inherit = "account.move"
+
+    invoice_warn_msg = fields.Text(compute="_compute_invoice_warn_msg")
+
+    def _compute_invoice_warn_msg(self):
```
Comment (MiquelRForgeFlow) [inline account_invoice_warn_message/models/account_move.py:16]:
```suggestion
    @api.depends('type', 'state', 'partner_id.invoice_warn', 'partner_id.parent_id.invoice_warn')
    def _compute_invoice_warn_msg(self):
```

🤔

File: account_invoice_warn_message/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        for rec in self:
+            if rec.partner_id.parent_id:
+                p = rec.partner_id.parent_id
```
Comment (MiquelRForgeFlow) [inline account_invoice_warn_message/models/account_move.py]:
```suggestion
            p = rec.partner_id.commercial_partner_id
```

🤔

File: account_invoice_warn_message/tests/test_account_invoice_warn_message.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                                "quantity": 1,
+                                "price_unit": 42,
+                            },
+                        ),
+                    ],
+                }
+            )
+        )
+        self.assertEqual(
+            invoice.invoice_warn_msg, self.warn_msg + "\n" + self.warn_msg_parent
```
Comment (AdriaGForgeFlow) [inline account_invoice_warn_message/tests/test_account_invoice_warn_message.py]: Given on how you compute the `invoice_warn_msg`, this should be:
```python
invoice.invoice_warn_msg, self.warn_msg_parent + "\n" + self.warn_msg
```

**Review by HaraldPanten — APPROVED** [review]
Hi @HviorForgeFlow ,

Functional review. 👍  Could you check comments, please?

Thx!

File: account_invoice_warn_message/README.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
+=============================
```
Comment (ValentinVinagre) [inline account_invoice_warn_message/README.rst:1]: the readme don't be separate in different files?

File: account_invoice_warn_message/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2020 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (ValentinVinagre) [inline account_invoice_warn_message/__manifest__.py:2]: add https in all documents.

File: account_invoice_warn_message/models/account_move.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                    and rec.partner_id.parent_id.invoice_warn == "warning"
+                ):
+                    rec.invoice_warn_msg = rec.partner_id.parent_id.invoice_warn_msg
+                    if rec.partner_id.invoice_warn == "warning":
+                        rec.invoice_warn_msg += "\n%s" % rec.partner_id.invoice_warn_msg
+                        continue
+                elif rec.partner_id.invoice_warn == "warning":
+                    rec.invoice_warn_msg = rec.partner_id.invoice_warn_msg
+                    continue
+            rec.invoice_warn_msg = False
```
Comment (ValentinVinagre) [inline account_invoice_warn_message/models/account_move.py:34]: why don't use 'commercial_partner_id' field?
Why do you use 'continue'? Better to use a variable res and write this value to the register at the end. ¿no?

**Review by ValentinVinagre — COMMENTED** [review]
Functional review 👍, some comments.
I have an IMP, when you create a sales invoice, in the wizard it does not show any warning, would it be interesting to show it? because if not, the user only sees the warning once the invoice has been created.

**Comment by HaraldPanten** [comment]
Hi @HviorForgeFlow are you still working on this?

THX!

**Pull request merged**